### ISSUE TITLE
Extracting multi layer exr

### ISSUE THREAD

**Issue by kirang4e** [MrKepzie/Natron#415]
Hello,
Iam working on a 3D comp.
3 passes rendered in exr.
When I load it in Natron,It's not representing actual texture/color.
I could not understand how to extract them using shuffle node.Its not showing any layers inside.
(This shuffle node is confusing.A simple tut will be great.)
I selected passes In read node itself, but still its not reading correctly..showing blue dots.

![ntrexr](https://cloud.githubusercontent.com/assets/10396656/5853097/d8e3edaa-a245-11e4-9c21-3f13867ba86e.png)

It is working perfectly in nuke..

![nukeexr](https://cloud.githubusercontent.com/assets/10396656/5853099/df88a826-a245-11e4-94e3-eaaa763e1db5.png)

Here is the exr,which I used.
http://www.mediafire.com/download/lpbg7sv7lf7hlz2/art021.zip

Thank you.

**Comment by MrKepzie** [comment]
The blue channel seems heavily wrecked. I’ll try to see what happens, where did you produce that file  ?

**Comment by devernay** [comment]
It seems to be an OpenImageIO bug (I checked with OIIO's image viewer, iv, which give the same glitches).
Probably because some components are float, and others are half, maybe @lgritz knows better about this.

Can you try to make the same image with all components float and/or all components half?

If it solves the issue, I'll file an OIIO bug.

```
chopper.exr:

file format version: 2, flags 0x200
cameraAperture (type float): 20.12
cameraFarClip (type float): 1e+30
cameraFarRange (type float): 1000
cameraFov (type float): 48.2144
cameraNearClip (type float): 0.1
cameraNearRange (type float): 0
cameraProjection (type int): 0
cameraTargetDistance (type float): 0
cameraTransform (type m44f):
   (0.71934 0.694658 1.49012e-08 0
    0.248943 -0.257788 0.933581 0
    0.64852 -0.671562 -0.358368 0
    13.4352 -17.5011 -5.71905 1)
channels (type chlist):
    A, 32-bit floating-point, sampling 1 1
    B, 32-bit floating-point, sampling 1 1
    Diffuse.A, 16-bit floating-point, sampling 1 1
    Diffuse.B, 16-bit floating-point, sampling 1 1
    Diffuse.G, 16-bit floating-point, sampling 1 1
    Diffuse.R, 16-bit floating-point, sampling 1 1
    G, 32-bit floating-point, sampling 1 1
    R, 32-bit floating-point, sampling 1 1
    Z Depth.B, 16-bit floating-point, sampling 1 1
    Z Depth.G, 16-bit floating-point, sampling 1 1
    Z Depth.R, 16-bit floating-point, sampling 1 1
compression (type compression): zip, multi-scanline blocks
dataWindow (type box2i): (0 0) - (1279 719)
displayWindow (type box2i): (0 0) - (1279 719)
gamma (type float): 1
lineOrder (type lineOrder): increasing y
pixelAspectRatio (type float): 1
screenWindowCenter (type v2f): (0 0)
screenWindowWidth (type float): 1
tiles (type tiledesc):
    single level
    tile size 64 by 64 pixels
type (type string): "tiledimage"

```

**Comment by kirang4e** [comment]
@MrKepzie  Actually I downloaded it from a tutorial page,so we don't have much details about it's origin. :(
@devernay  I checked metadata in nuke,It's 16-bit half float. I extracted them into separate layers and rendered individually. It's working :+1:   . I checked meta data on individual layers,they all 16-bit half float only.
![ntrexrfix](https://cloud.githubusercontent.com/assets/10396656/5854054/ac686be8-a24f-11e4-81d9-03e7cce8f6e9.png)

How can I separate them in Natron??

**Comment by blackearth2014** [comment]
Try creating three separate read nodes with the same file. Then change the channel components out. You should be able to see all three each connected to the viewer.

**Comment by kirang4e** [comment]
That I already did in 1st screenshot. please have a look :)

**Comment by devernay** [comment]
@kirang4e Nuke's metadata is wrong. The correct metadata is above (obtained using the *official* utility exrheader). This is very probably a bug with OpenImageIO.

**Comment by kirang4e** [comment]
@devernay Ok.Thank you. I learnt a small thing, exrheader ;).
I'm looking for another issue.How to extract multi layers in Natron? how to shuffle copy one pass to comp flow?

**Comment by MrKepzie** [comment]
As we did not yet implement multi-plane OpenFX suite, the only possible way for now is:

- Create multiple Read nodes to open the same file and read 4 different channels from each Read node
- Shuffle each Read node if needed

Don't worry internally, the same file will be read only once, it's just the pixel shuffling that will arise a bit more.

**Comment by kirang4e** [comment]
Ok.Thanks.Congrats for the Award :) 
Can we expect multi-plane OpenFX suite in next version?

**Comment by MrKepzie** [comment]
Not sure but it's surely in the roadmap though

**Comment by devernay** [comment]
the bug was fixed in OIIO, see https://github.com/OpenImageIO/oiio/issues/1038

**Comment by prasanth3d** [comment]
hi which node can i use in Natron for the 'expression' node in nuke?Please help

**Comment by CameraRick** [comment]
SeExprSimple should do the trick
